Robust.Server/GameStates/PVSSystem.cs

```diff
@@ -551,18 +551,24 @@ public void ReturnToPool((Dictionary<EntityUid, MetaDataComponent> metadata, Rob
     {
         //are we valid?
         //sometimes uids gets added without being valid YET (looking at you mapmanager) (mapcreate & gridcreated fire before the uids becomes valid)
-        if (!node.Value.IsValid()) return;
 
-        //did we already get added?
-        if (toSend.ContainsKey(node.Value)) return;
+        // As every map is parented to uid 0 in the tree we still need to get their children, plus because we go top-down
+        // we may find duplicate parents with children we haven't encountered before
+        // on different chunks (this is especially common with direct grid children)
+        if (node.Value.IsValid() && !toSend.ContainsKey(node.Value))
+        {
+            //are we new?
+            var (entered, budgetFail) = ProcessEntry(in node.Value, seenSet, previousVisibleEnts, ref newEntitiesSent,
+                ref totalEnteredEntities);
 
-        //are we new?
-        var (entered, budgetFail) = ProcessEntry(in node.Value, seenSet, previousVisibleEnts, ref newEntitiesSent,
-            ref totalEnteredEntities);
+            if (budgetFail) return;
 
-        if (budgetFail) return;
+            AddToSendSet(in node.Value, metaDataCache[node.Value], toSend, fromTick, entered);
+        }
+
+        //did we already get added?
+        // if (toSend.ContainsKey(node.Value)) return;
```

Comment:
Moved all of this inside the if statement so it only cares about adding to set if it's never been seen, otherwise it just goes onto getting the children.